### PR TITLE
[Practices] compliant reporting mark ineligible flow

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Dropdown to specify one or more ineligibility reasons for a client (mostly ported over from the prototype). A change in status also affects colors throughout the compliant reporting module. Also supports an "other" reason with a free text field

Screen Shot 2022-03-24 at 4 34 51 PM

Screen Shot 2022-03-24 at 4 34 20 PM

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Related issues

> Closes [#11619](https://github.com/Recidiviz/recidiviz-data/issues/11619)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by ohaibbq — APPROVED** [review]

File: src/core/PracticesClientProfile/CompliantReportingModule/CompliantReportingDenial.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      <Wrapper>
+        <Dropdown>
+          <StatusAwareButton {...buttonProps}>{buttonText}</StatusAwareButton>
+          <DropdownMenu>
+            {Object.entries(REASONS_MAP).map(([code, desc]) => (
+              <DropdownMenuItem
+                key={code}
+                onClick={() => {
+                  client.setCompliantReportingDenialReasons(
+                    xor(reasons, [code]).sort()
```
Comment (Catacola) [inline src/core/PracticesClientProfile/CompliantReportingModule/CompliantReportingDenial.tsx:129]: nice use of `xor`

File: src/firestore/types.ts [hunk 2]
```diff
@@ -98,23 +98,24 @@ export type CompliantReportingEligibleRecord = {
...
+  otherReason?: string;
+  updated: UpdateLog;
+};
+
+type CompliantReportingUpdateRecord = {
+  denial?: CompliantReportingDenial;
+  referralForm?: {
+    updated: UpdateLog;
   };
+  completed?: UpdateLog;
```
Comment (Catacola) [inline src/firestore/types.ts:120]: Why does `referralForm` have a nested `updated` field but the `completed` one is top-level?

**Review by Catacola — APPROVED** [review]
Just a few minor checks and snarky comments.

File: src/firestore/types.ts (hunk 2 above)
Comment (macfarlandian) [reply src/firestore/types.ts:120]: actually I will address this in the next PR that builds out this functionality

File: src/core/PracticesClientProfile/CompliantReportingModule/CompliantReportingDenial.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import { rem } from "polished";
+import React from "react";
+import styled from "styled-components/macro";
+
+import { OTHER_KEY } from "../../../PracticesStore";
+import { ClientProfileProps } from "../types";
+import { STATUS_COLORS, useStatusColors } from "./utils";
+
+const REASONS_MAP = {
+  DECF: "No effort to pay fine and costs",
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply src/core/PracticesClientProfile/CompliantReportingModule/CompliantReportingDenial.tsx:39]: this is what's in the document, but I asked and will update later if needed

**Pull request merged**